src/go/k8s/cmd/configurator/main.go

```diff
@@ -159,20 +179,129 @@ func main() {
 }
 
 func initializeSeedSeverList(
-	cfg *config.Config, c configuratorConfig, index brokerID,
+	cfg *config.Config, c configuratorConfig, index brokerID, k8sClient client.Client,
 ) error {
+	if index != 0 {
+		return nil
+	}
+
 	empty, err := IsRedpandaDataFolderEmpty(c.dataDirPath)
 	if err != nil {
 		return fmt.Errorf("checking Redpanda data folder content (%s): %w", c.dataDirPath, err)
 	}
 
-	if index == 0 && empty {
+	if !empty {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	stsName := renderStsName(c.hostName, index)
+
+	sts := &v1.StatefulSet{}
+	err = k8sClient.Get(ctx, client.ObjectKey{
+		Name: stsName,
+	}, sts)
+	if err != nil {
+		return fmt.Errorf("retrieving statful set: %w", err)
+	}
+
+	if *sts.Spec.Replicas == 1 {
 		cfg.Redpanda.SeedServers = []config.SeedServer{}
+		return nil
+	}
+
+	podList := &corev1.PodList{}
+	err = k8sClient.List(ctx, podList, &client.ListOptions{
+		LabelSelector: k8sLabels.SelectorFromSet(map[string]string{
+			labels.InstanceKey: stsName,
+		}),
+	})
+	if err != nil {
+		return fmt.Errorf("listing available PODs: %w", err)
+	}
+
+	anyRedpandaPodIsRunning := false
+	for _, pod := range podList.Items {
+		if pod.Name == c.hostName {
+			continue
+		}
+		if pod.Status.Phase == corev1.PodRunning {
```

Comment:
I understand this is a problem we need to avoid, but I wonder if we can find a different way. I've seen e.g. cases where all nodes crash during an upgrade, which can still lead the first one to think he's alone. Isn't Redpanda supposed to handle these situations?

Reply:
apparently Redpanda might form 2 clusters -> https://github.com/redpanda-data/redpanda/issues/333